src/Docker/Runner/DataLoader/DataLoader.php

```diff
@@ -464,7 +464,9 @@ public function cleanWorkspace(): void
 
     private function isReusableWorkspace(StagingProviderInterface $stagingProvider): bool
     {
-        return $stagingProvider->getStaging()->getType() === AbsWorkspaceStaging::getType()
-            || $stagingProvider->getStaging()->getType() === RedshiftWorkspaceStaging::getType();
+        return $this->getStagingStorageInput() === InputStrategyFactory::WORKSPACE_ABS ||
+            $this->getStagingStorageOutput() === InputStrategyFactory::WORKSPACE_ABS ||
+            $this->getStagingStorageInput() === InputStrategyFactory::WORKSPACE_REDSHIFT ||
+            $this->getStagingStorageOutput() === InputStrategyFactory::WORKSPACE_REDSHIFT;
```

Comment:
Tak jsem si to prosel jeste jednou a podruhy a moc teda nevim. Cela ta metoda `DataLoader::cleanupWorkspace()` byla pred tim v podstate uplne nezavisla na komponente (nebo cemkoliv jinym, vyjma `$this->configId`) - pro kazdou komponentu zkusila provest cleanup vsech znamych typu staging. Nove se cleanup nedela, pokud komponenta nema input/output na workspace abs/redshift.

Jako svym zpusobem mi to ted prijde lepsi - dela se to podle komponenty, ktera bezi. Ale spis je mi vlastne divny, proc se cely ten cleanup dela tak genericky a ne jen podle staging, ktery ta komponenta pouziva? :thinking:

A jeste me trochu zarazi, to to nefailnulo zadny testy :slightly_smiling_face: